org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/wizard/template/AbstractProjectTemplate.java

```diff
@@ -0,0 +1,122 @@
+/*******************************************************************************
+ * Copyright (c) 2017 itemis AG (http://www.itemis.de) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.ui.wizard.template;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.log4j.Logger;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+
+/**
+ * A template definition for a project, used by the new project wizard. Defines the UI (label, description, icon, variables) on how to
+ * present the template to the user and a list or {@link ProjectFile} elements to be generated by the wizard when the template is selected.
+ * The {@link ProjectFile}'s can contain references to {@link ProjectVariable}'s to allow the user to configure the generated code.
+ * 
+ * Instances of this class are generated from classes annotated with {@link ProjectTemplate}.
+ * 
+ * @author Arne Deutsch - Initial contribution and API
+ * @since 2.14
+ */
+public abstract class AbstractProjectTemplate {
```

Comment:
Should this be called `ProjectTemplate` instead? The fact that it's an abstract class is not necessarily smth that should be carried in the name.

Reply:
Agree to that. But I have used the name "ProjectTemplate" for the (active) annotation already. Hence I have to use a different name for the super class. The active annotation automatically extends the annotated class from that one. For that reason it seems ok for be to use "Abstract" as prefix. "ProjectTemplateImpl" would be an alternative but from my point of view not better in any way.

Reply:
Ok